pkg/interfaces/contracts/vault/IBaseRouter.sol

```diff
@@ -0,0 +1,257 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { IERC4626 } from "@openzeppelin/contracts/interfaces/IERC4626.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import { IVault } from "./IVault.sol";
+
+import { AddLiquidityKind, RemoveLiquidityKind, SwapKind } from "./VaultTypes.sol";
+
+interface IBaseRouter {
```

Comment:
This could also be `IRouterQueries`, except that you'll also need the data structs afterwards right?

Do you think it'd be worth splitting this into `IRouterQueries` and `RouterTypes`, like we do with the vault, or is it too much?

Reply:
At first I thought that separating Types and Queries into two different contracts would be an overkill, since no contract would call "RouterTypes" separated from IRouterQueries, but thinking twice it can improve the UX from developers integrating with Balancer routers.